packages/devtools_app/lib/src/screens/memory/allocation_profile_table_view.dart

```diff
@@ -0,0 +1,424 @@
+  ];
+
+  static final _vmDeveloperModeColumns = [
+    FieldInstanceCountColumn(heap: HeapGeneration.newSpace),
+    FieldSizeColumn(heap: HeapGeneration.newSpace),
+    FieldInternalSizeColumn(heap: HeapGeneration.newSpace),
+    FieldExternalSizeColumn(heap: HeapGeneration.newSpace),
+    FieldInstanceCountColumn(heap: HeapGeneration.oldSpace),
+    FieldSizeColumn(heap: HeapGeneration.oldSpace),
+    FieldInternalSizeColumn(heap: HeapGeneration.oldSpace),
+    FieldExternalSizeColumn(heap: HeapGeneration.oldSpace),
+  ];
+
+  final AllocationProfileTableViewController allocationProfileController;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<AllocationProfile?>(
+      valueListenable: allocationProfileController.currentAllocationProfile,
+      builder: (context, profile, _) {
+        // TODO(bkonyi): make this an overlay so the table doesn't
+        // disappear when we're retrieving new data, especially since the
+        // spinner animation freezes when retrieving the profile.
+        if (profile == null) {
+          return const CenteredCircularProgressIndicator();
+        }
+        return ValueListenableBuilder<bool>(
+          valueListenable: preferences.vmDeveloperModeEnabled,
+          builder: (context, vmDeveloperModeEnabled, _) {
+            return ElevatedCard(
+              padding: const EdgeInsets.only(),
+              child: LayoutBuilder(
+                builder: (context, constraints) {
+                  return FlatTable<ClassHeapStats?>(
+                    columnGroups: [
+                      ..._columnGroups,
+                      if (vmDeveloperModeEnabled)
+                        ..._vmDeveloperModeColumnGroups,
+                    ],
+                    columns: [
+                      ..._columns,
+                      if (vmDeveloperModeEnabled) ..._vmDeveloperModeColumns,
+                    ],
+                    data: profile.members!.where(
+                      (element) {
+                        return element.bytesCurrent != 0 ||
+                            element.newSpace.externalSize != 0 ||
+                            element.oldSpace.externalSize != 0;
+                      },
+                    ).toList(),
+                    sortColumn: _columns.first,
+                    sortDirection: SortDirection.ascending,
+                    onItemSelected: (item) => null,
+                    keyFactory: (d) => Key(d!.classRef!.name!),
+                  );
+                },
+              ),
+            );
+          },
+        );
+      },
+    );
+  }
+}
+
+class _AllocationProfileTableControls extends StatelessWidget {
+  const _AllocationProfileTableControls({
+    Key? key,
+    required this.allocationProfileController,
+    this.refreshOnGcKey,
+    this.refreshKey,
+  }) : super(key: key);
+
+  final AllocationProfileTableViewController allocationProfileController;
+  final Key? refreshOnGcKey;
+  final Key? refreshKey;
```

Comment:
Yes, they're referenced in the test.

Reply:
You can lookup the button by other attributes like icon, label, tooltip, or type (RefreshOnGCToggleButton). I'd recommend doing this instead of creating and passing around keys just for the sake of testing